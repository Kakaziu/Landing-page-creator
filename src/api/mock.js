export const pageData = [
  {
    id: 1,
    attributes: {
      title: 'PÃ¡gina 1',
      slug: 'page-1',
      footer_text: [
        {
          type: 'paragraph',
          children: [
            {
              type: 'text',
              text: 'footer',
            },
          ],
        },
      ],
      sections: [
        {
          id: 1,
          __component: 'section.section-two-columns',
          title: 'Section two columns',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          image: {
            data: {
              id: 1,
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/drplqaxi0/image/upload/v1704251425/pexels_angel_rondon_19483671_99d3895603.jpg',
              },
            },
          },
          button: {
            id: 1,
            text_button: 'Clique aqui',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 1,
            name: 'section-two-columns-1',
            section_id: 'hero',
            background: true,
          },
        },
        {
          id: 2,
          __component: 'section.section-two-columns',
          title: 'Sobre',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          image: {
            data: {
              id: 1,
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/drplqaxi0/image/upload/v1704251425/pexels_angel_rondon_19483671_99d3895603.jpg',
              },
            },
          },
          button: null,
          metadata: {
            id: 2,
            name: 'section-two-columns-2',
            section_id: 'sobre',
            background: false,
          },
        },
        {
          id: 1,
          __component: 'section.section-grid',
          title: 'grid',
          description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
          button: {
            id: 3,
            text_button: 'aaaaa clica',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 3,
            name: 'grid-1',
            section_id: 'depoimentos',
            background: true,
          },
          text_grid: [
            {
              id: 1,
              title: 'grid 1',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
            {
              id: 2,
              title: 'grid 2',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
            {
              id: 3,
              title: 'grid 3',
              description:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
            },
          ],
          image_grid: [],
        },
        {
          id: 1,
          __component: 'section.section-content',
          title: 'contato',
          content: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ',
                },
                {
                  type: 'text',
                  text: 'ipsum dolor sit amet',
                  bold: true,
                },
                {
                  type: 'text',
                  text: ', consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu gittis mattis.',
                },
              ],
            },
          ],
          button: {
            id: 4,
            text_button: 'botao',
            url: '#',
            size: 'big',
          },
          metadata: {
            id: 4,
            name: 'section-content',
            section_id: 'contato',
            background: true,
          },
        },
        {
          id: 1,
          __component: 'section.section-box',
          title: 'Section Box',
          button: null,
          paragraph: [
            {
              id: 1,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
            {
              id: 2,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
            {
              id: 3,
              text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
            },
          ],
          section_metadata: {
            id: 5,
            name: 'section_box',
            section_id: 'vvvvvv',
            background: false,
          },
        },
      ],
      menu: {
        id: 1,
        logo_title: 'logo',
        logo_link: '#',
        image_logo: {
          data: {
            id: 1,
            attributes: {
              url: 'a.svg',
            },
          },
        },
        menu_links: [
          {
            id: 1,
            text: 'Home',
            url: '#home',
            open_in_new_tab: false,
          },
          {
            id: 2,
            text: 'Sobre',
            url: '#sobre',
            open_in_new_tab: false,
          },
          {
            id: 3,
            text: 'Depoimentos',
            url: '#depoimentos',
            open_in_new_tab: false,
          },
          {
            id: 4,
            text: 'Contato',
            url: '#contato',
            open_in_new_tab: false,
          },
        ],
      },
    },
  },
];

export const dataMenu = {
  id: 1,
  logo_title: 'logo',
  logo_link: '#',
  image_logo: {
    data: {
      id: 1,
      attributes: {
        url: 'a.svg',
      },
    },
  },
  menu_links: [
    {
      id: 1,
      text: 'Home',
      url: '#home',
      open_in_new_tab: false,
    },
    {
      id: 2,
      text: 'Sobre',
      url: '#sobre',
      open_in_new_tab: false,
    },
    {
      id: 3,
      text: 'Depoimentos',
      url: '#depoimentos',
      open_in_new_tab: false,
    },
    {
      id: 4,
      text: 'Contato',
      url: '#contato',
      open_in_new_tab: false,
    },
  ],
};

export const menuLinks = [
  {
    id: 1,
    text: 'Home',
    url: '#home',
    open_in_new_tab: false,
  },
  {
    id: 2,
    text: 'Sobre',
    url: '#sobre',
    open_in_new_tab: false,
  },
  {
    id: 3,
    text: 'Depoimentos',
    url: '#depoimentos',
    open_in_new_tab: false,
  },
  {
    id: 4,
    text: 'Contato',
    url: '#contato',
    open_in_new_tab: false,
  },
];

export const sectionTwoColumnsMock = {
  id: 1,
  __component: 'section.section-two-columns',
  title: 'Section two columns',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu porttitor augue, ut auctor sem. Cras rhoncus pellentesque ante non pharetra. Maecenas pretium consectetur sapien pellentesque rutrum. Aliquam vitae tempor purus, at rhoncus orci. Suspendisse a sem non dolor sagittis mattis.',
  image: {
    data: {
      id: 1,
      attributes: {
        alternativeText: 'image',
        url: 'b.svg',
      },
    },
  },
  button: {
    id: 1,
    text_button: 'Clique aqui',
    url: '#',
    size: 'big',
  },
  metadata: {
    id: 1,
    name: 'section-two-columns-1',
    section_id: 'hero',
    background: true,
  },
};

export const sectionBoxMock = {
  id: 1,
  __component: 'section.section-box',
  title: 'Section Box',
  button: null,
  paragraph: [
    {
      id: 1,
      text: 'paragraph',
    },
    {
      id: 2,
      text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
    },
    {
      id: 3,
      text: "here are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.",
    },
  ],
  section_metadata: {
    id: 5,
    name: 'section_box',
    section_id: 'vvvvvv',
    background: false,
  },
};
