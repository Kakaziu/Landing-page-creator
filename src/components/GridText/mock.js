export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  gridText: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'small',
  },
};
