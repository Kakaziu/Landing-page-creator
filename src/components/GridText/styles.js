import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.xlarge};
  margin-top: ${theme.spacings.huge};
  width: 100%;

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
  }
`}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${
        background === 1 ? theme.colors.white : theme.colors.primaryColor
      }
    }
  `}
`;
