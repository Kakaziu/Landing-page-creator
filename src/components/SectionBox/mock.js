export default {
  title: 'Section Box',
  paragraphs: [
    {
      text: 'There are 1',
    },
    {
      text: 'There are 2',
    },
    {
      text: 'There are 3',
    },
  ],
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'small',
  },
};
