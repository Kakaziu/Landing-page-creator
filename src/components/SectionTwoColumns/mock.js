export default {
  title: 'Grid two columns',
  text: 'Alo hamigos',
  srcImg: './assets/images/image1.svg',
  button: {
    textButton: 'Clique aqui',
    urlButton: '#',
    size: 'big',
  },
};
