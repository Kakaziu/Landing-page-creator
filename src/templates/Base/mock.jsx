import links from '../../components/NavLink/mock';
import GridText from '../../components/GridText';
import gridMock from '../../components/GridText/mock';

export default {
  children: (
    <>
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
      <GridText {...gridMock} background />
      <GridText {...gridMock} />
    </>
  ),
  links: links,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste de footer</p>',
};
