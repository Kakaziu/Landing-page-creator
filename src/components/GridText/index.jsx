import P from 'prop-types';
import SectionBackground from '../SectionBackground';
import * as Styled from './styles';
import Heading from '../Heading';
import Button from '../Button';
import TextComponent from '../TextComponent';

const GridText = ({
  title,
  description,
  grid = [],
  background = false,
  button,
}) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container background={+background}>
        <Styled.TitleContainer>
          <Heading uppercase>{title}</Heading>
          <TextComponent>{description}</TextComponent>
        </Styled.TitleContainer>
        <Styled.GridContainer>
          {grid.map((element) => (
            <Styled.GridElement key={element.title}>
              <Heading size="small">{element.title}</Heading>
              <TextComponent>{element.description}</TextComponent>
            </Styled.GridElement>
          ))}
        </Styled.GridContainer>
        {button && (
          <Button url={button.urlButton} size={button.size}>
            {button.textButton}
          </Button>
        )}
      </Styled.Container>
    </SectionBackground>
  );
};

GridText.propTypes = {
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    }),
  ),
  background: P.bool,
  button: Button.propTypes,
};

export default GridText;
