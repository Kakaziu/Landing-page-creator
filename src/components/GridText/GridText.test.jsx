import GridText from '.';
import mock from './mock';
import renderTheme from '../../styles/render-theme';
import { screen } from '@testing-library/react';

describe('<GridText/>', () => {
  it('should render grids', () => {
    renderTheme(<GridText {...mock} />);

    expect(screen.getAllByRole('heading', { name: /teste/i })).toHaveLength(
      mock.grid.length,
    );
  });

  it('should not render grids', () => {
    renderTheme(<GridText />);

    expect(screen.queryAllByRole('heading', { name: /teste/i })).toHaveLength(
      0,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<GridText {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
