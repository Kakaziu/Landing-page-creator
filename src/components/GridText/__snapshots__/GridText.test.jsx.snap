// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText/> should match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  color: #0A1128;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c9 {
  color: #0A1128;
  text-transform: none;
  font-size: 2.4rem;
}

.c11 {
  display: block;
}

.c12 {
  font-weight: bold;
  border: none;
  background-color: #dc143c;
  color: #FFFFFF;
  border-radius: 4px;
  padding: 8px 20px;
  font-size: 1.6rem;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.c12:hover {
  background-color: #b81132;
}

.c8 .c4 {
  margin-bottom: 1.6rem;
}

.c3 .c4 {
  margin-bottom: 1.6rem;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 4.0rem;
  margin-top: 5.6rem;
  width: 100%;
}

.c2 .c4 {
  margin-bottom: 1.6rem;
  color: #0A1128;
}

.c2 .c10 {
  margin-top: 2.4rem;
}

.c6 {
  font-size: 1.6rem;
}

@media (max-width:768px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    background="0"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        background="0"
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
            isdark="1"
            uppercase="1"
          >
            My grid
          </h1>
          <p
            class="c6"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </p>
        </div>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <h1
              class="c4 c9"
              isdark="1"
              uppercase="0"
            >
              Teste 1
            </h1>
            <p
              class="c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h1
              class="c4 c9"
              isdark="1"
              uppercase="0"
            >
              Teste 2
            </h1>
            <p
              class="c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
          <div
            class="c8"
          >
            <h1
              class="c4 c9"
              isdark="1"
              uppercase="0"
            >
              Teste 3
            </h1>
            <p
              class="c6"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </p>
          </div>
        </div>
        <a
          class="c10 c11"
          href="#"
        >
          <button
            class="c12"
          >
            Clique aqui
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
